BizTravel360/src/main/java/com/travel/BizTravel360/transport/TransportService.java

```diff
@@ -2,56 +2,159 @@
-    public void createTransport(Transport transport) throws Exception {
-        List<Transport> transports = readAllTransports();
-        long maxId = transports.stream().mapToLong(Transport::getTransportId).max().orElse(0);
-        transport.setTransportId(maxId + 1);
-        transports.add(transport);
-        fileService.writerToFile(transports, transportFilePath);
+    public Transport saveTransport(Transport transport) throws IOException {
+        try {
+            formatTransportDates(transport);
+            trimTransport(transport);
+            validateTransport(transport);
+            
+            transport.setTransportId((UUID.randomUUID().getMostSignificantBits() & Long.MAX_VALUE));
+            transports.add(transport);
+            fileService.writerToFile(transports, transportFilePath);
+            formatTransportDates(transport);
+            log.info("Transport {}", transport);
+            return transport;
+        }catch (IOException e){
+            log.error("Failed to save transport {}", transport);
+            throw new RuntimeException("Failed to save transport: " + transport);
+        }
+    }
+    
+    @Override
+    public List<Transport> fetchTransportList() {
+        List<Transport> reverseTransports = new ArrayList<>(this.transports);
+        Collections.reverse(reverseTransports);
+        return new ArrayList<>(reverseTransports);
+    }
+    
+    @Override
+    public Transport updateTransport(Transport updateTransport, Long transportId) throws IOException {
+        Transport transportToUpdate = findTransportById(transportId);
+        trimTransport(updateTransport);
+        
+        if (transportToUpdate != null) {
+            transportToUpdate.setTypeTransport(updateTransport.getTypeTransport());
+            transportToUpdate.setTransportIdentifier(updateTransport.getTransportIdentifier());
+            transportToUpdate.setDeparture(updateTransport.getDeparture());
+            transportToUpdate.setDepartureDateTime(updateTransport.getDepartureDateTime());
+            transportToUpdate.setArrival(updateTransport.getArrival());
+            transportToUpdate.setArrivalDateTime(updateTransport.getArrivalDateTime());
+            transportToUpdate.setCount(updateTransport.getCount());
+            return transportToUpdate;
+        }else {
+            return null;
+        }
     }
     
     @Override
-    public List<Transport> readAllTransports(){
+    public void deleteTransportById(Long transportId) throws IOException {
+        transports.removeIf(t -> Objects.equals(t.getTransportId(), transportId));
+        persistTransport();
+    }
+    
+    @Override
+    public Transport findTransportById(Long transportId) throws IOException {
+        return transports.stream()
+                .filter(t -> Objects.equals(t.getTransportId(), transportId))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("No transport with id " + transportId));
```

Comment:
```suggestion
                .orElseThrow(() -> new NotFoudExceptiom(String.format("No transport with id:  %s",  transportId)));
```
utworzyć nową klasę i katolog exceptions, TransportNotFoudExceptiom